.github/workflows/workflow.yml

```diff
@@ -0,0 +1,142 @@
+name: Colcon Build, Test, Coverage and Autoformat
+
+on:
+  pull_request:
+    branches:
+      - dev
+      - pre-release
+      - release
+
+  workflow_dispatch:
+
+jobs:
+  Autoformat:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@main
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 2
+
+      - name: Clang-format style check
+        uses: THI-Drone/action-clang-format@main
+
+      - name: Autopep8 style check
+        uses: THI-Drone/action-autopep@main
+
+
+  Build-and_test:
+    needs: Autoformat
+    runs-on: self-hosted
+
+    steps:
+
+    - name: Get the date
+      id: date
+      run: echo "date=$(date +'%F_%H-%M-%S')" >> $GITHUB_OUTPUT
+
+    - name: Setup Docker Container 
+      run: | 
+        CONTAINER_NAME="ros-container"
+        IMAGE="ros:humble"
+        OLD_ID=$(docker inspect --format "{{.Id}}" $IMAGE)
+        docker pull $IMAGE
+        NEW_ID=$(docker inspect --format "{{.Id}}" $IMAGE)
+        INSTALL_DEPENDENCIES=false
+        
+        if [ "$OLD_ID" != "$NEW_ID" ]; then
+          echo "update container"
+          docker stop $CONTAINER_NAME || true
+          docker rm $CONTAINER_NAME || true
+          docker run --name $CONTAINER_NAME -i -d $IMAGE bash
+          INSTALL_DEPENDENCIES=true
+        else
+          echo "use old container"
+          CONTAINER_ID=$(docker ps -a -q --filter name="^/${CONTAINER_NAME}$")
+
+          if [ -z "$CONTAINER_ID" ]; then
+            echo "run new container"
+            docker run --name $CONTAINER_NAME  -i -d $IMAGE bash
+            INSTALL_DEPENDENCIES=true
+          else
+            RUNNING_CONTAINER_ID=$(docker ps -q --filter name="^/${CONTAINER_NAME}$")
+
+            if [ -z "$RUNNING_CONTAINER_ID" ]; then
+                docker start $CONTAINER_NAME
+                docker exec -i $CONTAINER_NAME /bin/bash -c "rm -rf /thi-drone-ws/"
+            else
+                echo "Container $CONTAINER_NAME is already running."
+                docker exec -i $CONTAINER_NAME /bin/bash -c "rm -rf /thi-drone-ws/"
+            fi
+          fi
+        fi
+        if [ "$INSTALL_DEPENDENCIES" = true ]; then
+          docker exec -i $CONTAINER_NAME /bin/bash -c "apt-get update && apt-get install -y nlohmann-json3-dev && apt-get install -y python3-pip && pip install setuptools==58.2.0 && pip install pytest && rosdep update && pip install coverage && pip install colcon-coveragepy-result && pip install pytest-cov && apt-get install ros-humble-ament-cmake -y && apt-get install ros-humble-ament-cmake-clang-format -y && apt-get install ros-humble-ament-cmake-cppcheck -y && apt-get install ros-humble-ament-cmake-cpplint -y && apt-get install ros-humble-ament-cmake-flake8 -y && apt-get install ros-humble-ament-cmake-lint-cmake -y && apt-get install ros-humble-ament-cmake-pep257 -y && apt-get install gcc lcov -y && apt-get install python3-colcon-lcov-result && apt-get install lcov -y"
```

Comment:
Gleiches Problem wie beim demo_package

Reply:
Hier eben das gleiche. Würde das Repo so erstmal mit nem PR in alle submopdule mergen und dann wenn jemand/ich die kapazitäten frei hat es anpassen. Und über distribute_workflow morgen einmal mit allen beteiligten reden.